packages/client/src/modules/play/playerActions/TeamActionsContent.tsx

```diff
@@ -6,42 +6,40 @@ import { useState } from "react";
 
 import { Typography } from "../../common/components/Typography";
 import { TeamAction } from "../../../utils/types";
-import { useCurrentStep } from "../context/playContext";
+import { usePlay, useTeamActions } from "../context/playContext";
 import { Accordion } from "../../common/components/Accordion";
-import { CURRENT_YEAR_COUNTRY_POWER_NEED_IN_GW, getEnergy } from "../constants";
+import { CURRENT_YEAR_COUNTRY_POWER_NEED_IN_GW } from "../constants";
 import { t } from "../../translations";
 import { Icon } from "../../common/components/Icon";
 import { Slider } from "../../common/components/Slider";
-import { TEAM_ACTIONS_MOCKS } from "../constants/mocks";
-import { computeProductionEnergyStats } from "../utils/production";
+import { computeTeamActionStats } from "../utils/production";
 import { Dialog } from "../../common/components/Dialog";
 
 export { TeamActionsContent };
 
 function TeamActionsContent({ style }: { style?: React.CSSProperties }) {
-  const currentStep = useCurrentStep();
+  const { teamActionsAtCurrentStep } = useTeamActions();
 
   const [openHelpDialog, setOpenHelpDialog] = useState(false);
   const [helpCardLink, setHelpCardLink] = useState("");
 
-  const energiesToDisplay = getEnergy({ stepId: currentStep?.id });
-  const energyNameToTeamActions = Object.fromEntries(
-    TEAM_ACTIONS_MOCKS.map((teamAction) => [teamAction.action.name, teamAction])
-  );
-
   return (
     <>
       <Box style={style}>
         <Accordion
-          options={energiesToDisplay.map((energy) =>
-            createTeamActionOption({
-              teamAction: energyNameToTeamActions[energy.name],
-              onOpenHelpCard: () => {
-                setHelpCardLink(energy.helpCardLink);
-                setOpenHelpDialog(true);
-              },
-            })
-          )}
+          options={teamActionsAtCurrentStep
+            .map((teamAction) =>
+              createTeamActionOption({
+                teamAction,
+                onOpenHelpCard: () => {
+                  setHelpCardLink(teamAction.action.helpCardLink);
+                  setOpenHelpDialog(true);
+                },
+              })
+            )
+            .filter((option): option is NonNullable<typeof option> =>
```

Comment:
ça fait quoi ça ?

Reply:
Typescript ne détecte pas quand on filtre les objets non nuls dans un `filter`. Le type guard `option is ...` permet d'assurer à typescript que les éléments restants sont bien non nuls